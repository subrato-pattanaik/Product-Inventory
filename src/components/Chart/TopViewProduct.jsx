import React from "react";
import { Doughnut } from "react-chartjs-2";

const data = {
  labels: ["Samsung Refigerator", "Lenovo Desktop", "Nikon Camera"],
  datasets: [
    {
      data: [300, 50, 100],
      backgroundColor: ["#FF6384", "#36A2EB", "#FFCE56"],
      hoverBackgroundColor: ["#FF6384", "#36A2EB", "#FFCE56"],
    },
  ],
};

function TopViewProduct() {
  return (
    <div>
      <h4>Top 3 Products</h4>
      <hr className="bg-warning mb-3" />
      <Doughnut data={data} width={20} height={20} />
    </div>
  );
}

export default TopViewProduct;
