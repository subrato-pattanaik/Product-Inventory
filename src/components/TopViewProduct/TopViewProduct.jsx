import React from "react";
import { withRouter } from "react-router-dom";
function TopViewProduct() {
  return (
    <div>
      <h1>this is tvp section</h1>
      <p>this is tvp section</p>
      <p>this is tvp section</p>
      <p>this is tvp section</p>
    </div>
  );
}

export default withRouter(TopViewProduct);
