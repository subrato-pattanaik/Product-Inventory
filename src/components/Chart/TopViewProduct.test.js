import React from "react";
import { shallow } from "enzyme";

import TopViewProduct from "./TopViewProduct";

describe("Testing of TopViewProduct Page", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<TopViewProduct />);
  });

  it("should test snapshot of TopViewProduct page", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should test header built-in component", () => {
    expect(wrapper.find("h4").exists()).toEqual(true);
  });

  it("should test hr using classname", () => {
    expect(wrapper.find(".bg-warning")).toHaveLength(1);
  });
});
