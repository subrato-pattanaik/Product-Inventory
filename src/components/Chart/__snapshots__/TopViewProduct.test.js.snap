// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing of TopViewProduct Page should test snapshot of TopViewProduct page 1`] = `
<div>
  <h4>
    Top 3 Products
  </h4>
  <hr
    className="bg-warning mb-3"
  />
  <Doughnut
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": Array [
              "#FF6384",
              "#36A2EB",
              "#FFCE56",
            ],
            "data": Array [
              300,
              50,
              100,
            ],
            "hoverBackgroundColor": Array [
              "#FF6384",
              "#36A2EB",
              "#FFCE56",
            ],
          },
        ],
        "labels": Array [
          "Samsung Refigerator",
          "Lenovo Desktop",
          "Nikon Camera",
        ],
      }
    }
    height={20}
    width={20}
  />
</div>
`;
